pkg/plugins/golang/v3/scaffolds/internal/templates/makefile.go

```diff
@@ -130,7 +130,7 @@ run: manifests generate fmt vet ## Run a controller from your host.
 
 .PHONY: docker-build
 docker-build: test ## Build docker image with the manager.
-	docker build -t ${IMG} .
+	docker build -t ${IMG} --build-arg GOARCH=$(shell go env GOARCH) .
```

Comment:
I do not think that we can change in this way.

Note that: 
I am Operator Author who is working in darwin/amd64 local env. However, my operator will be published on Linux/amd64 cluster which is the default/common scenario. So if we move with your change it would fail because the GOARCH used is not the default. 

I think we can: 
Create a variable in the Makefile with the GOARCH fixed for amd64 
then we can comment on top to let authors know if they want to build the Operator manager image for another arch then can update that.

Reply:
Okay, I can create 2 variables for the target OS and ARCH and set them by default at linux and amd64:
```
TARGET_GOOS   ?= linux
TARGET_GOARCH ?= amd64
```
Then they can be passed as build arguments to the Dockerfile:
`docker build -t ${IMG} --build-arg TARGET_GOOS=${TARGET_GOOS} --build-arg TARGET_GOARCH=${TARGET_GOARCH} .`